copy/single.go

```diff
@@ -365,26 +369,39 @@ func (ic *imageCopier) compareImageDestinationManifestEqual(ctx context.Context,
 		return nil, nil
 	}
 
+	compressionAlgos := set.New[string]()
+	for _, srcInfo := range ic.src.LayerInfos() {
+		compression := compressionAlgorithmFromMIMEType(srcInfo)
+		compressionAlgos.Add(compression.Name())
+	}
+
+	algos, err := compression.AlgorithmsByNames(compressionAlgos.Values())
+	if err != nil {
+		return nil, err
+	}
+
 	// Destination and source manifests, types and digests should all be equivalent
 	return &copySingleImageResult{
-		manifest:         destManifest,
-		manifestMIMEType: destManifestType,
-		manifestDigest:   srcManifestDigest,
+		manifest:              destManifest,
+		manifestMIMEType:      destManifestType,
+		manifestDigest:        srcManifestDigest,
+		compressionAlgorithms: algos,
 	}, nil
 }
 
 // copyLayers copies layers from ic.src/ic.c.rawSource to dest, using and updating ic.manifestUpdates if necessary and ic.cannotModifyManifestReason == "".
-func (ic *imageCopier) copyLayers(ctx context.Context) error {
+func (ic *imageCopier) copyLayers(ctx context.Context) ([]compressiontypes.Algorithm, error) {
+	compressionAlgos := set.New[string]()
```

Comment:
Absolutely non-blocking: This can be set only just before it’s computed (around `diffIDs := `)